plugins/sqlfluff-templater-dbt/sqlfluff_templater_dbt/templater.py

```diff
@@ -416,32 +416,27 @@ def _unsafe_process(self, fname, in_str=None, config=None):
         # turn is used by our parent class' (JinjaTemplater) slice_file()
         # function.
         old_from_string = Environment.from_string
-        try:
-            make_template = None
-
-            def from_string(*args, **kwargs):
-                """Replaces (via monkeypatch) the jinja2.Environment function."""
-                nonlocal make_template
-                # Is it processing the node corresponding to fname?
-                globals = kwargs.get("globals")
-                if globals:
-                    model = globals.get("model")
-                    if model:
-                        if model.get("original_file_path") == original_file_path:
-                            # Yes. Capture the important arguments and create
-                            # a make_template() function.
-                            env = args[0]
-                            globals = args[2] if len(args) >= 3 else kwargs["globals"]
-
-                            def make_template(in_str):
-                                env.add_extension(SnapshotExtension)
-                                return env.from_string(in_str, globals=globals)
-
-                return old_from_string(*args, **kwargs)
-
-        finally:
-            # Undo the monkeypatch.
-            Environment.from_string = from_string
```

Comment:
The comment was wrong here. This line was **applying** the monkeypatch. I did some reorg so the line that **removes** the monkeypatch is in a `try/finally` block.